### ISSUE TITLE
Function macro can be defined but not used

### ISSUE THREAD

**Issue by Tawling** [Zezombye/overpy#7]
[Examples used the demo for compilation]

When simply defining a function macro, compilation works properly.
```py
#!define foo(x)    bar = x

#!define bar A
#!define baz B

@Rule "Without function macro"
@Event global
bar = baz
```

However, if you attempt to use the function macro, compilation fails.
```py
#!define foo(x)    bar = x

#!define bar A
#!define baz B

@Rule "With function macro"
@Event global
foo(baz)
```

This fails with console error:
```
Error: line 8, col 1: No match found for keyword 'x'
    at error (utils.js:557)
    at translate (utils.js:292)
    at tows (utils.js:306)
    at parse (overpyCompiler.js:611)
    at parseAssignment (overpyCompiler.js:1136)
    at parse (overpyCompiler.js:478)
    at compileRule (overpyCompiler.js:437)
    at compile (overpyCompiler.js:37)
    at demoCompile (demo:280)
    at HTMLButtonElement.onclick (demo:40)
```

Tokenization seems to fail to replace the argument(s) prior to parse/compile.

**Comment by Zezombye** [comment]
It's working properly, as lowercase x is not a valid keyword :p

The macros themselves get replaced (which is why it's evaluating "x").

**Comment by Zezombye** [comment]
Nevermind, I didn't notice that x was the macro argument name.

Should be fixed in 1.2
